doc/NAC/PnacSonicHLD.md

```diff
@@ -0,0 +1,374 @@
+### 3.2.7 Sequence of Events flowchart
+Following Flow chart describes actions in respective sequence after events like port UP/DOWN takes place.
+![PNAC Use Case](images/PNAC_SOE.png)
+
+## 3.3 DB Changes
+As part of PNAC feature introduction following are changes in respective DB.
+### 3.3.1 ConfigDB Schema Changes
+This section describes incremental changes to ConfigDB implemented =for PNAC feature.
+Feature configuration:
+```
+"FEATURE": {
+           "nac": {
+                "auto_restart": "enabled",            
+                "has_global_scope": "True",            
+                "has_per_asic_scope": "False",            
+                "has_timer": "False",            
+                "high_mem_alert": "disabled",            
+                "set_owner": "local",            
+                "state": "enabled"            
+           },
+```        
+***NAC Global configuration***
+```
+"NAC":{
+        "global": {        
+            "admin_state": "up",            
+            "auth_type": "local",            
+            "nac_type": "port"            
+        }        
+    },
+```
+***NAC_Session Table*** 
+```
+"NAC_SESSION": {
+        "Ethernet0": {        
+            "admin_state": "down",            
+            "nac_status": "unauthorized"            
+        },        
+        "Ethernet4": {        
+            "admin_state": "up",            
+            "nac_status": "unauthorized"
+```
+### 3.3.2 App DB changes
+1. ***nac_status*** field is added to the NAC_SESSION table.
+2. ***admin_state*** filed is added to the NAC_SESSION table.
+3. At Global level a flag is maintained to enable or disable PNAC feature
+## 3.4 CLI Commands
+#
+| S.No. |  CLI Command                                     | Description                                                                  |
+|--------|----------------------------------------------|---------------------------------------------------------------|
+|   1         | $config nac <enable\|disable>       | This CLI option allows a user to enable\|disable the NAC functionality globally in the system.     |    
+| 2   | $config nac interface <enable\|disable> <interface_name\|all> | This CLI option allows a user to configure the NAC feature control at interface level.|
+|3          | $show nac | This CLI option displays the NAC global information configured at Switch |
+|4         | $show nac interface <interface_name\|all> | This CLI option displays the NAC information configured at all ports of the switch |
+| 5       | $config nac type <port\|mac> | This CLI Option shall configure the type of NAC to be configured/used across the SONiC switch. ***Note*** This CLI shall be applicable only if NAC feature is enabled, globally. |
+
+# 4 Serviceability and Debug
+Existing debug framework is extended for PNAC feature by adding more debug logs for appropriate conditions.
+# 5 Warm Reboot
+Warm reboot is not impacting functionality of the feature.
```

Comment:
During the test plan review it was stated that client traffic is disrupted during a Warm Reboot. The above statement is contradiction. If client traffic is disrupted, PNAC essentially does not support Warm Reboot.

Reply:
Ok got it will update HLD to reflect the right behavior.